File: crates/semantiq-parser/tests/fixtures/php/sample.php
<?php
namespace App\Service;

use Foo\Bar;

class UserService {
    const VERSION = "1.0";

    public function greet(string $name): string {
        return "Hi $name";
    }
}

interface Greeter {
    public function greet(): string;
}

trait Loggable {
    public function log(string $msg): void {}
}

enum Status {
    case Active;
    case Inactive;
}

function helper(): int {
    return 1;
}

File: crates/semantiq-parser/tests/fixtures/csharp/sample.cs
using System;

namespace MyApp.Services {
    public class UserService {
        private List<User> users;

        public void AddUser(User user) {
            users.Add(user);
        }
    }

    public struct User {
        public int Id;
    }

    public interface IGreeter {
        string Greet();
    }

    public enum Status {
        Active,
        Inactive
    }
}
